pkg/versiongetter/github_release.go

```diff
@@ -89,15 +94,30 @@ func (g *GitHubReleaseVersionGetter) List(ctx context.Context, pkg *registry.Pac
 					Description: release.GetBody(),
 					URL:         release.GetHTMLURL(),
 				}
+				prevCnt = len(items)
 				items = append(items, &fuzzyfinder.Item{
 					Item:    tagName,
 					Preview: fuzzyfinder.PreviewVersion(v),
 				})
 			}
 		}
+		if limit > 0 && len(items) >= limit { // Reach the limit
+			if len(items) > limit {
+				items = items[:limit]
+			}
+			return items, nil
+		}
 		if len(releases) != opt.PerPage {
 			return items, nil
 		}
+		// After filtering, not enough versions added.
+		// Increase per_page to reduce the consumption of GitHub API.
+		if opt.PerPage < ghMaxPerPage && len(items)-prevCnt < opt.PerPage {
+			opt.PerPage *= 2
+			if opt.PerPage > ghMaxPerPage {
+				opt.PerPage = ghMaxPerPage
+			}
+		}
```

Comment:
Changing PerPage in the middle of loop causes the wrong result because ListReleases gets releases `Page * PerPage ~ ((Page +1) * PerPage) - 1`.

For example, if the Original Perpage is `30` and Page is `0`,
ListReleases gets `0 ~ 29`, then if you change PerPage to `30*2 = 60` and increment Page to `1`,
ListReleases gets `60 ~ 119`, so releases `30 ~ 59` are lost.